src/toxcore/dht/packet/ping_request.rs

```diff
@@ -72,6 +86,28 @@ impl PingRequest {
             payload,
         }
     }
+
+    // Root parser for PingRequest object
+    fn parse<'a, E: ParseError<&'a [u8]>>(i: &'a [u8]) ->IResult<&'a [u8], PingRequest, E> {
```

Comment:
This function is very verbose. We used nom3's do_parse to make code compact. I would like you to implement the same in nom5.

Reply:
It seems it can be simplified to:

```rust
map(tuple((
    tag("\x00"),
    PublicKey::from_bytes5,
    Nonce::from_bytes5,
    map(rest_5, |bytes: &[u8]| bytes.to_vec()),
)), |(_, pk, nonce, payload)| PingRequest { pk, nonce, payload })
```

What do you think about this option? Honestly I'd prefer plain functions rather than using weird looking dsl (which also can't be formatted with rustfmt).

Reply:
`do_parse` is still there https://docs.rs/nom/5.0.0-beta1/nom/macro.do_parse.html
So nevermind, I'm fine using it - it should be easier to migrate.